src/pages/index.js

```diff
@@ -56,48 +94,100 @@ const addCardPopUp = new PopupWithForm(
-/*                                   Section                                  */
-/* -------------------------------------------------------------------------- */
+/* ---------------------------Popup Confirm Delete--------------------------- */
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      console.log(item);
-      const cardEl = renderCard(item);
-      cardSection.addItem(cardEl);
-    },
-  },
-  ".cards__list"
+const confirmDeletePopup = new PopupWithConfirmation(
+  "#confirm-image-delete-popup",
+  handleDeleteConfirmSubmit
 );
-cardSection.renderItems();
+confirmDeletePopup.setEventListeners();
 
-const userInfoNew = new UserInfo(".profile__name", ".profile__description");
+/* ------------------------Change Profile Avatar Popup----------------------- */
+
+const changeProfileAvatarPopUp = new PopupWithForm(
+  "#change-profile-avatar-popup",
+  handleChangeProfileAvatarFormSubmit
+);
+changeProfileAvatarPopUp.setEventListeners();
+
+const userInfoNew = new UserInfo(
+  profileName,
+  profileDescription,
+  profileAvatar
+);
+
+api.fetchUserInfo().then((data) => {
+  let userData = data;
+  userInfoNew.setUserInfo(userData.name, userData.about);
+  userInfoNew.setUserAvatar(userData.avatar);
+});
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 
 function renderCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteConfirmSubmit,
+    handleCardLike
+  );
   return card.getView();
 }
 
 function handleImageClick(data) {
-  console.log(data);
   imagePopUp.open(data);
 }
 
 function handleEditProfileFormSubmit(data) {
+  profileEditPopup.setLoading('Saving...');
   userInfoNew.setUserInfo(data);
-  profileEditPopup.close();
+  api.editUserInfo(data).then(() => {
+    profileEditPopup.close();
+  }).finally(() => profileEditPopup.setLoading('Save'));
+}
+
+function handleChangeProfileAvatarFormSubmit(data) {
+  changeProfileAvatarPopUp.setLoading('Saving...')
+  userInfoNew.setUserAvatar(data.link);
+  api.editUserAvatar(data.link).then(() => {
+    changeProfileAvatarPopUp.close();
+  }).finally(() => changeProfileAvatarPopUp.setLoading('Save'));
 }
 
 function handleAddCardFormSubmit(data) {
+  addCardPopUp.setLoading('Saving...');
   const cardValue = renderCard(data);
   cardSection.addItem(cardValue);
-  addCardPopUp.close();
+  api
+    .addCard(data)
+    .then(() => {
+      addCardPopUp.close();
+    })
+    .finally(() => addCardPopUp.setLoading('Create'));
+}
+
+function handleDeleteConfirmSubmit(card) {
+  confirmDeletePopup.open();
+  confirmDeletePopup.setSubmitAction(() => {
```

Comment:
Needs correction: you should also notify the user about loading process by changing the submit button text